molecule/default/tests/test_default.rb

```diff
@@ -1,3 +1,17 @@
 describe directory('/etc/') do
   its('owner') { should eq 'root' }
 end
+
+describe file('/var/lib/pulp/database/pulp.sqlite3') do
+  it { should exist }
+  its('owner') { should eq 'pulp' }
+  its('group') { should eq 'pulp' }
+  its('size') { should > 300 }
```

Comment:
What's the meaning of this particular test?

Reply:
I don't think its a very good test, but that what was done before.
We can decide not to add it (we don't have many tests to review anyway). But we should have a discussion about what does it mean a good state of pulp deployment, such as:
- service is running
- existence or absence of log entries
- database entries, database structure
- successful command execution (client connection, server status,...)

Reply:
If SQLite is to be the default database, then making assertions on `/var/lib/pulp/database/pulp.sqlite3` is an OK thing to do. I'm specifically curious about this "size" assertion.

More broadly, it'd probably be good to think about making this installer default to PostgreSQL. Both the Pulp developers and Pulp QE have found that SQLite suffers from such low performance that Pulp can completely break.

Reply:
Regarding size  I think they tried to test that the database is not empty.
Tests of this type don't say much about the state of the system. Its a mean to a result, which should be taken care by Ansible in any case. Generally speaking, our verification should be Black-box testing.

"making PostgreSQL the default" - Can we open GitHub issues for this project?

Reply:
> Regarding size I think they tried to test that the database is not empty.

Would `its('size') { should > 0 }` work?

> "making PostgreSQL the default" - Can we open GitHub issues for this project?

The Pulp devs do all issue tracking on https://pulp.plan.io. See https://pulp.plan.io/issues/3716.

Reply:
> Would `its('size') { should > 0 }` work?
Don't know, maybe the file comes with initial content and that's why they set 300.

> https://pulp.plan.io/issues/3716
I'm not Pulp dev, just a humble contributer :sweat_smile:
If you wish, I can sign-up and assign tickets to myself and open new.
@lyrch, what do you think?

Reply:
>  Don't know, maybe the file comes with initial content and that's why they set 300.

Let's imagine that we merge this PR as-is, and a week later, this assertion starts failing. Nothing else goes wrong, just this assertion. What do we do? Do we drop the assertion? Do we try to make the assertion pass by putting content into the database, despite the fact that nothing else seems to have broken? If we don't know the meaning of an assertion, then we shouldn't add it to a test suite.

Also, who is "they?" I thought this was your PR. :wink:

Reply:
This is a *refactor* of the original test that existed in pulp3 role, before molecule was removed from it. So my vote is on removing the assertion.

Reply:
> This is a *refactor* of the original test that existed in pulp3 role, before molecule was removed from it.

Got it. I now understand what the commit message is trying to say. I didn't understand the meaning of the commit message prior to this clarification.

> So my vote is on removing the assertion.

Agreed. I like that this assertion was already in the test suite, but in the interest of only executing meaningful assertions, I also prefer to drop it.